### PR TITLE
adds device query info to debug session

### LINKED ISSUE #121: Do device-info query at start of debug session
Certain functionality requires knowledge of what firmware or debug session version we're running, so run a[ device-info](https://developer.roku.com/en-gb/docs/developer-program/debugging/external-control-api.md#general-ecp-commands) query at the start of a debug session and store that on the session so it's available for the lifetime of that debug session.

The request looks something like this:
`http://192.168.1.37:8060/query/device-info`

The results should be parsed from xml into a usable javascript object /json structure. 

Also create a typescript interface called `DeviceInfo` for all of the known fields. Just copy the info from here into a new interface: https://github.com/rokucommunity/vscode-brightscript-language/blob/100070dee1fe83d0144a7be40a3d55396cfb6e40/src/ActiveDeviceManager.ts#L240

### PR DESCRIPTION
**Pull request by sethmaclean**
This pull request offers changes for Fixes #121

### CODE REVIEW

File: src/debugSession/BrightScriptDebugSession.ts [hunk 1]
```diff
@@ -49,7 +49,12 @@ import type { AugmentedSourceBreakpoint } from '../managers/BreakpointManager';
 import { BreakpointManager } from '../managers/BreakpointManager';
 import type { LogMessage } from '../logging';
 import { logger, debugServerLogOutputEventTransport } from '../logging';
-import { waitForDebugger } from 'inspector';
+// import { waitForDebugger } from 'inspector';
+import type { DeviceInfo } from '../DeviceInfo';
+import axios from 'axios';
+// import { Console } from 'console';
+import * as xml2js from 'xml2js';
+// import { ThemeIcon } from 'vscode';
```
Comment (TwitchBronBron) [inline src/debugSession/BrightScriptDebugSession.ts:57]: Can you remove these commented-out imports?

File: src/debugSession/BrightScriptDebugSession.ts [hunk 2]
```diff
@@ -204,6 +247,15 @@ export class BrightScriptDebugSession extends BaseDebugSession {
...
         }
 
+        // fetches the device info and parses the xml data to JSON object
+        // try {
+        this.deviceInfo = await this.fetchDeviceInfo(this.launchConfiguration.host, this.launchConfiguration.remotePort);
+        // } catch (e) {
+        //     const errorMessage = `Could not fetch Roku device info`
+        //     this.showPopupMessage(errorMessage, 'error')
+        //     throw(e)
+        // }
```
Comment (TwitchBronBron) [inline src/debugSession/BrightScriptDebugSession.ts:257]: Remove this commented-out code

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]
Looks pretty good. just a few small changes.

File: src/DeviceInfo.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    'supports-airplay'?: boolean;
+    'has-play-on-roku'?: boolean;
+    'has-mobile-screensaver'?: boolean;
+    'support-url'?: string;
+    'grandcentral-version'?: string;
+    'trc-version'?: number;
+    'trc-channel-version'?: string;
+    'davinci-version'?: string;
+    'av-sync-calibration-enabled'?: number;
+    // Anything nre they might add that we do not know about
```
Comment (MilapNaik) [inline src/DeviceInfo.ts:69]:
```suggestion
    // Anything new they might add that we do not know about
```

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**